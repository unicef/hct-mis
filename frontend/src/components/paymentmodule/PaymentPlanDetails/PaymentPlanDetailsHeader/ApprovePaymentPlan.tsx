import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';
import * as Yup from 'yup';
import { Field, Form, Formik } from 'formik';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { DialogContainer } from '../../../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../../../containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '../../../../hooks/useSnackBar';
import { FormikTextField } from '../../../../shared/Formik/FormikTextField/FormikTextField';
import { LoadingButton } from '../../../core/LoadingButton';
import { GreyText } from '../../../core/GreyText';
import { usePaymentPlanAction } from '../../../../hooks/usePaymentPlanAction';
import { Action, PaymentPlanQuery } from '../../../../__generated__/graphql';

export interface ApprovePaymentPlanProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const ApprovePaymentPlan = ({
  paymentPlan,
}: ApprovePaymentPlanProps): React.ReactElement => {
  const { t } = useTranslation();
  const [approveDialogOpen, setApproveDialogOpen] = useState(false);
  const { showMessage } = useSnackbar();
  const {
    mutatePaymentPlanAction: approve,
    loading: loadingApprove,
  } = usePaymentPlanAction(
    Action.Approve,
    paymentPlan.id,
    () => showMessage(t('Payment Plan has been approved.')),
    () => setApproveDialogOpen(false),
  );
  const initialValues = {
    comment: '',
  };

  const validationSchema = Yup.object().shape({
    comment: Yup.string()
      .min(2, 'Too short')
      .max(255, 'Too long'),
  });

  const shouldShowLastApproverMessage = (): boolean => {
    const { approvalNumberRequired } = paymentPlan;
    const approvalsCount =
      paymentPlan.approvalProcess?.edges[0]?.node.actions.approval.length;

    return approvalNumberRequired - 1 === approvalsCount;
  };

  return (
    <>
      <Formik
        initialValues={initialValues}
        onSubmit={(values, { resetForm }) => {
          approve(values.comment);
          resetForm({});
        }}
        validationSchema={validationSchema}
      >
        {({ submitForm }) => (
          <>
            <Box p={2}>
              <Button
                color='primary'
                variant='contained'
                onClick={() => setApproveDialogOpen(true)}
                data-cy='button-approve'
              >
                {t('Approve')}
              </Button>
            </Box>
            <Dialog
              open={approveDialogOpen}
              onClose={() => setApproveDialogOpen(false)}
              scroll='paper'
              aria-labelledby='form-dialog-title'
              maxWidth='md'
            >
              <DialogTitleWrapper>
                <DialogTitle id='scroll-dialog-title'>
                  {t('Approve Payment Plan')}
                </DialogTitle>
              </DialogTitleWrapper>
              <DialogContent>
                <DialogContainer>
                  <Box p={5}>
                    {t('Are you sure you want to approve this Payment Plan?')}
                  </Box>
                  {shouldShowLastApproverMessage() && (
                    <Box p={5}>
                      <GreyText>
                        {t(
                          'Note: You are the last approver. Upon proceeding, this Payment Plan will be automatically moved to authorization stage.',
                        )}
                      </GreyText>
                    </Box>
                  )}
                  <Form>
                    <Field
                      name='comment'
                      multiline
                      fullWidth
                      variant='filled'
                      label='Comment (optional)'
                      component={FormikTextField}
                    />
                  </Form>
                </DialogContainer>
              </DialogContent>
              <DialogFooter>
                <DialogActions>
                  <Button onClick={() => setApproveDialogOpen(false)}>
                    CANCEL
                  </Button>
                  <LoadingButton
                    loading={loadingApprove}
                    type='submit'
                    color='primary'
                    variant='contained'
                    onClick={submitForm}
                    data-cy='button-submit'
                  >
                    {t('Approve')}
                  </LoadingButton>
                </DialogActions>
              </DialogFooter>
            </Dialog>
          </>
        )}
      </Formik>
    </>
  );
};
