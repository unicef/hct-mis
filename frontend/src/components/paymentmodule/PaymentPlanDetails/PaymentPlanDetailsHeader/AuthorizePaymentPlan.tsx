import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';
import * as Yup from 'yup';
import { Field, Form, Formik } from 'formik';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { DialogContainer } from '../../../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../../../containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '../../../../hooks/useSnackBar';
import { FormikTextField } from '../../../../shared/Formik/FormikTextField/FormikTextField';
import { LoadingButton } from '../../../core/LoadingButton';
import { GreyText } from '../../../core/GreyText';

export interface AuthorizePaymentPlanProps {
  paymentPlanId: string;
}

export const AuthorizePaymentPlan = ({
  paymentPlanId,
}: AuthorizePaymentPlanProps): React.ReactElement => {
  const { t } = useTranslation();
  const [authorizeDialogOpen, setAuthorizeDialogOpen] = useState(false);

  const { showMessage } = useSnackbar();
  // const [mutate] = useActivateCashPlanPaymentVerificationMutation();
  // const activate = async (): Promise<void> => {
  //   try {
  //     await mutate({
  //       variables: { cashPlanVerificationId },
  //       refetchQueries,
  //     });
  //   } catch (error) {
  //     /* eslint-disable-next-line no-console */
  //     console.log('error', error?.graphQLErrors);
  //     if (
  //       error?.graphQLErrors?.[0]?.validationErrors
  //         ?.activateCashPlanPaymentVerification?.phone_numbers
  //     ) {
  //       showMessage(
  //         error?.graphQLErrors?.[0]?.validationErrors?.activateCashPlanPaymentVerification?.phone_numbers.join(
  //           '\n',
  //         ),
  //       );
  //     } else {
  //       showMessage(t('Error during activating.'));
  //     }
  //   }

  //   showMessage(t('Verification plan has been activated.'));
  // };
  const initialValues = {
    comment: '',
  };

  const validationSchema = Yup.object().shape({
    comment: Yup.string()
      .min(2, 'Too short')
      .max(255, 'Too long'),
  });

  return (
    <>
      <Formik
        initialValues={initialValues}
        onSubmit={(values, { resetForm }) => {
          console.log('authorize');
          resetForm({});
        }}
        validationSchema={validationSchema}
      >
        {({ submitForm }) => (
          <>
            <Box p={2}>
              <Button
                color='primary'
                variant='contained'
                onClick={() => setAuthorizeDialogOpen(true)}
                data-cy='button-lock-plan'
              >
                {t('Authorize')}
              </Button>
            </Box>
            <Dialog
              open={authorizeDialogOpen}
              onClose={() => setAuthorizeDialogOpen(false)}
              scroll='paper'
              aria-labelledby='form-dialog-title'
              maxWidth='md'
            >
              <DialogTitleWrapper>
                <DialogTitle id='scroll-dialog-title'>
                  {t('Authorize')}
                </DialogTitle>
              </DialogTitleWrapper>
              <DialogContent>
                <DialogContainer>
                  <Box p={5}>
                    {t('Are you sure you want to authorize this Payment Plan?')}
                  </Box>
                  <Box p={5}>
                    <GreyText>
                      {t(
                        'Note: Upon Proceeding, this Payment Plan will be automatically moved to Finance Review stage.',
                      )}
                    </GreyText>
                  </Box>
                  <Form>
                    <Field
                      name='comment'
                      multiline
                      fullWidth
                      variant='filled'
                      label='Comment (optional)'
                      component={FormikTextField}
                    />
                  </Form>
                </DialogContainer>
              </DialogContent>
              <DialogFooter>
                <DialogActions>
                  <Button onClick={() => setAuthorizeDialogOpen(false)}>
                    CANCEL
                  </Button>
                  <LoadingButton
                    loading
                    type='submit'
                    color='primary'
                    variant='contained'
                    onClick={() => console.log(paymentPlanId)}
                    data-cy='button-submit'
                  >
                    {t('Authorize')}
                  </LoadingButton>
                </DialogActions>
              </DialogFooter>
            </Dialog>
          </>
        )}
      </Formik>
    </>
  );
};
