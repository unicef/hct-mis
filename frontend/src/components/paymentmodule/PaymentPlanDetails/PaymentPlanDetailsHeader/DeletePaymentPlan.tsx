import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton,
} from '@material-ui/core';
import { Delete } from '@material-ui/icons';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import {
  PaymentPlanQuery,
  useDeletePpMutation,
} from '../../../../__generated__/graphql';
import { DialogContainer } from '../../../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../../../containers/dialogs/DialogTitleWrapper';
import { useBaseUrl } from '../../../../hooks/useBaseUrl';
import { getPaymentPlanUrlPart } from '../../../../utils/utils';
import { LoadingButton } from '../../../core/LoadingButton';

export interface DeletePaymentPlanProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const DeletePaymentPlan = ({
  paymentPlan,
}: DeletePaymentPlanProps): React.ReactElement => {
  const { t } = useTranslation();
  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
  const { baseUrl } = useBaseUrl();
  const history = useHistory();
  const [mutate, { loading: loadingDelete }] = useDeletePpMutation();
  const { id } = paymentPlan;

  const handleDelete = (): void => {
    mutate({
      variables: {
        paymentPlanId: id,
      },
    });
    history.push(
      `/${baseUrl}/payment-module/${getPaymentPlanUrlPart(
        paymentPlan.isFollowUp,
      )}/`,
    );
  };

  return (
    <>
      <Box p={2}>
        <IconButton
          data-cy='button-delete-payment-plan'
          onClick={() => setDeleteDialogOpen(true)}
        >
          <Delete />
        </IconButton>
      </Box>
      <Dialog
        open={deleteDialogOpen}
        onClose={() => setDeleteDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle>{t('Delete Payment Plan')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t('Are you sure you want to delete this Payment Plan?')}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button
              data-cy='button-cancel'
              onClick={() => setDeleteDialogOpen(false)}
            >
              CANCEL
            </Button>
            <LoadingButton
              loading={loadingDelete}
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => handleDelete()}
              data-cy='button-submit'
            >
              {t('Delete')}
            </LoadingButton>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
};
