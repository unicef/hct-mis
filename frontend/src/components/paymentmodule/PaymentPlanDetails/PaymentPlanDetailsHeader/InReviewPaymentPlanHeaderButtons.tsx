import React, { useState } from 'react';
import styled from 'styled-components';
import { Button } from '@material-ui/core';
import { EditRounded, Delete, FileCopy } from '@material-ui/icons';
import { useTranslation } from 'react-i18next';
import { ErrorButton } from '../../../core/ErrorButton';

const IconContainer = styled.span`
  button {
    color: #949494;
    min-width: 40px;
    svg {
      width: 20px;
      height: 20px;
    }
  }
`;

const ButtonContainer = styled.span`
  margin: 0 ${({ theme }) => theme.spacing(2)}px;
`;

export interface InReviewPaymentPlanHeaderButtonsProps {
  setEditState: Function;
  canDuplicate: boolean;
  canRemove: boolean;
  canEdit: boolean;
  canLock: boolean;
}

export function InReviewPaymentPlanHeaderButtons({
  setEditState,
  canDuplicate,
  canEdit,
  canLock,
  canRemove,
}: InReviewPaymentPlanHeaderButtonsProps): React.ReactElement {
  const { t } = useTranslation();
  const [openApprove, setOpenApprove] = useState(false);
  const [openDuplicate, setOpenDuplicate] = useState(false);
  const [openDelete, setOpenDelete] = useState(false);
  return (
    <div>
      {canLock && (
        <ButtonContainer>
          <ErrorButton onClick={() => setOpenApprove(true)}>
            {t('Reject')}
          </ErrorButton>
        </ButtonContainer>
      )}
      {canLock && (
        <ButtonContainer>
          <Button
            variant='contained'
            color='primary'
            onClick={() => setOpenApprove(true)}
          >
            {t('Mark as Reviewed')}
          </Button>
        </ButtonContainer>
      )}
    </div>
  );
}
