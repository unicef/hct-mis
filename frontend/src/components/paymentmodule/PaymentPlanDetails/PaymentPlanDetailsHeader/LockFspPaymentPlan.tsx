import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { DialogContainer } from '../../../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../../../containers/dialogs/DialogTitleWrapper';
import { usePaymentPlanAction } from '../../../../hooks/usePaymentPlanAction';
import { useSnackbar } from '../../../../hooks/useSnackBar';
import { Action, PaymentPlanQuery } from '../../../../__generated__/graphql';
import { LoadingButton } from '../../../core/LoadingButton';

export interface LockFspPaymentPlanProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const LockFspPaymentPlan = ({
  paymentPlan,
}: LockFspPaymentPlanProps): React.ReactElement => {
  const { t } = useTranslation();
  const { showMessage } = useSnackbar();
  const [lockDialogOpen, setLockDialogOpen] = useState(false);
  const {
    mutatePaymentPlanAction: lock,
    loading: loadingLock,
  } = usePaymentPlanAction(
    Action.LockFsp,
    paymentPlan.id,
    () => showMessage(t('Payment Plan FSPs are locked.')),
    () => setLockDialogOpen(false),
  );

  return (
    <>
      <Box p={2}>
        <Button
          color='primary'
          variant='contained'
          onClick={() => setLockDialogOpen(true)}
          data-cy='button-lock-plan'
        >
          {t('Lock FSP')}
        </Button>
      </Box>
      <Dialog
        open={lockDialogOpen}
        onClose={() => setLockDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>{t('Lock FSP')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t(
                'After you lock the FSP in this Payment Plan, you will be able to send the Payment Plan for approval.',
              )}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setLockDialogOpen(false)}>CANCEL</Button>
            <LoadingButton
              loading={loadingLock}
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => lock()}
              data-cy='button-submit'
            >
              {t('Lock FSP')}
            </LoadingButton>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
};
