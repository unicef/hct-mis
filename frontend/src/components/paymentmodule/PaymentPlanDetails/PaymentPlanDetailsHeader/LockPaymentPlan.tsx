import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { DialogContainer } from '../../../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../../../containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '../../../../hooks/useSnackBar';
import { Missing } from '../../../core/Missing';

export interface LockPaymentPlanProps {
  paymentPlanId: string;
}

export const LockPaymentPlan = ({
  paymentPlanId,
}: LockPaymentPlanProps): React.ReactElement => {
  const { t } = useTranslation();
  const [lockDialogOpen, setLockDialogOpen] = useState(false);

  const { showMessage } = useSnackbar();
  // const [mutate] = useActivateCashPlanPaymentVerificationMutation();
  // const activate = async (): Promise<void> => {
  //   try {
  //     await mutate({
  //       variables: { cashPlanVerificationId },
  //       refetchQueries,
  //     });
  //   } catch (error) {
  //     /* eslint-disable-next-line no-console */
  //     console.log('error', error?.graphQLErrors);
  //     if (
  //       error?.graphQLErrors?.[0]?.validationErrors
  //         ?.activateCashPlanPaymentVerification?.phone_numbers
  //     ) {
  //       showMessage(
  //         error?.graphQLErrors?.[0]?.validationErrors?.activateCashPlanPaymentVerification?.phone_numbers.join(
  //           '\n',
  //         ),
  //       );
  //     } else {
  //       showMessage(t('Error during activating.'));
  //     }
  //   }

  //   showMessage(t('Verification plan has been activated.'));
  // };
  return (
    <>
      <Box p={2}>
        <Button
          color='primary'
          variant='contained'
          onClick={() => setLockDialogOpen(true)}
          data-cy='button-lock-plan'
        >
          {t('Lock')}
        </Button>
      </Box>
      <Dialog
        open={lockDialogOpen}
        onClose={() => setLockDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            {t('Lock Payment Plan')}
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t(
                'After you lock this Payment Plan, you will be able to run entitlement formula for selected target population.',
              )}
            </Box>
            <Box p={5}>
              {t('Note:')} {t('There are')} <Missing />{' '}
              {t(
                'households missing payment channel information. Grievance tickets will be automatically created.',
              )}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setLockDialogOpen(false)}>CANCEL</Button>
            <Button
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => console.log(paymentPlanId)}
              data-cy='button-submit'
            >
              {t('Lock')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
};
