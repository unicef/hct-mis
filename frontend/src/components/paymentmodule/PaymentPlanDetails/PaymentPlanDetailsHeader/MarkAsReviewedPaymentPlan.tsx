import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';
import { Field, Form, Formik } from 'formik';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import * as Yup from 'yup';
import { DialogContainer } from '../../../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../../../containers/dialogs/DialogTitleWrapper';
import { usePaymentPlanAction } from '../../../../hooks/usePaymentPlanAction';
import { useSnackbar } from '../../../../hooks/useSnackBar';
import { FormikTextField } from '../../../../shared/Formik/FormikTextField/FormikTextField';
import { Action, PaymentPlanQuery } from '../../../../__generated__/graphql';
import { GreyText } from '../../../core/GreyText';
import { LoadingButton } from '../../../core/LoadingButton';

export interface MarkAsReviewedPaymentPlanProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const MarkAsReviewedPaymentPlan = ({
  paymentPlan,
}: MarkAsReviewedPaymentPlanProps): React.ReactElement => {
  const { t } = useTranslation();
  const [markAsReviewedDialogOpen, setMarkAsReviewedDialogOpen] = useState(
    false,
  );
  const { showMessage } = useSnackbar();
  const {
    mutatePaymentPlanAction: review,
    loading: loadingReview,
  } = usePaymentPlanAction(
    Action.Review,
    paymentPlan.id,
    () => showMessage(t('Payment Plan has been marked as reviewed.')),
    () => showMessage(t('Error during marking Payment Plan as reviewed.')),
    () => setMarkAsReviewedDialogOpen(false),
  );

  const shouldShowLastReviewerMessage = (): boolean => {
    const { financeReviewNumberRequired } = paymentPlan;
    const financeReviewsCount =
      paymentPlan.approvalProcess?.edges[0]?.node.actions.financeReview.length;

    return financeReviewNumberRequired - 1 === financeReviewsCount;
  };

  const initialValues = {
    comment: '',
  };

  const validationSchema = Yup.object().shape({
    comment: Yup.string()
      .min(2, 'Too short')
      .max(255, 'Too long'),
  });

  return (
    <>
      <Formik
        initialValues={initialValues}
        onSubmit={(values, { resetForm }) => {
          review(values.comment);
          resetForm({});
        }}
        validationSchema={validationSchema}
      >
        {({ submitForm }) => (
          <>
            <Box p={2}>
              <Button
                color='primary'
                variant='contained'
                onClick={() => setMarkAsReviewedDialogOpen(true)}
                data-cy='button-lock-plan'
              >
                {t('Mark as reviewed')}
              </Button>
            </Box>
            <Dialog
              open={markAsReviewedDialogOpen}
              onClose={() => setMarkAsReviewedDialogOpen(false)}
              scroll='paper'
              aria-labelledby='form-dialog-title'
              maxWidth='md'
            >
              <DialogTitleWrapper>
                <DialogTitle id='scroll-dialog-title'>
                  {t('Mark as Reviewed')}
                </DialogTitle>
              </DialogTitleWrapper>
              <DialogContent>
                <DialogContainer>
                  <Box p={5}>
                    {t(
                      'Are you sure you want to mark this Payment Plan as reviewed?',
                    )}
                  </Box>
                  {shouldShowLastReviewerMessage() && (
                    <Box p={5}>
                      <GreyText>
                        {t(
                          'Note: You are the last reviewer. Upon proceeding, this Payment Plan will be automatically moved to accepted status',
                        )}
                      </GreyText>
                    </Box>
                  )}
                  <Form>
                    <Field
                      name='comment'
                      multiline
                      fullWidth
                      variant='filled'
                      label='Comment (optional)'
                      component={FormikTextField}
                    />
                  </Form>
                </DialogContainer>
              </DialogContent>
              <DialogFooter>
                <DialogActions>
                  <Button onClick={() => setMarkAsReviewedDialogOpen(false)}>
                    CANCEL
                  </Button>
                  <LoadingButton
                    loading={loadingReview}
                    type='submit'
                    color='primary'
                    variant='contained'
                    onClick={submitForm}
                    data-cy='button-submit'
                  >
                    {t('Mark as reviewed')}
                  </LoadingButton>
                </DialogActions>
              </DialogFooter>
            </Dialog>
          </>
        )}
      </Formik>
    </>
  );
};
