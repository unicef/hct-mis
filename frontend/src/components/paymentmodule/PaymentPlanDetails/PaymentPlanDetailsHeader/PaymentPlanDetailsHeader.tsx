import { Box } from '@material-ui/core';
import styled from 'styled-components';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { hasPermissions, PERMISSIONS } from '../../../../config/permissions';
import { BreadCrumbsItem } from '../../../core/BreadCrumbs';
import { PageHeader } from '../../../core/PageHeader';
import {
  paymentPlanStatusMapping,
  paymentPlanStatusToColor,
} from '../../../../utils/utils';
import { StatusBox } from '../../../core/StatusBox';
import { PaymentPlanQuery } from '../../../../__generated__/graphql';
import { OpenPaymentPlanHeaderButtons } from './HeaderButtons/OpenPaymentPlanHeaderButtons';
import { LockedPaymentPlanHeaderButtons } from './HeaderButtons/LockedPaymentPlanHeaderButtons';
import { InApprovalPaymentPlanHeaderButtons } from './HeaderButtons/InApprovalPaymentPlanHeaderButtons';
import { AcceptedPaymentPlanHeaderButtons } from './HeaderButtons/AcceptedPaymentPlanHeaderButtons';
import { InAuthorizationPaymentPlanHeaderButtons } from './HeaderButtons/InAuthorizationPaymentPlanHeaderButtons';
import { InReviewPaymentPlanHeaderButtons } from './HeaderButtons/InReviewPaymentPlanHeaderButtons';

const StatusWrapper = styled.div`
  width: 140px;
  margin-left: 30px;
`;

interface PaymentPlanDetailsHeaderProps {
  businessArea: string;
  permissions: string[];
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const PaymentPlanDetailsHeader = ({
  businessArea,
  permissions,
  paymentPlan,
}: PaymentPlanDetailsHeaderProps): React.ReactElement => {
  const { t } = useTranslation();
  const [editState, setEditState] = useState(false);
  const breadCrumbsItems: BreadCrumbsItem[] = [
    {
      title: t('Payment Module'),
      to: `/${businessArea}/payment-module/`,
    },
  ];

  const canRemove = true;
  const canEdit = true;
  const canLock = true;
  const canSendForApproval = true;
  const canDuplicate = true;
  const canReject = true;
  const canApprove = true;
  const canAuthorize = true;
  const canMarkAsReviewed = true;
  const canDownloadXlsx = true;
  const canSendToFsp = true;

  let buttons;
  switch (paymentPlan.status) {
    case 'OPEN':
      buttons = (
        <>
          <OpenPaymentPlanHeaderButtons
            paymentPlan={paymentPlan}
            setEditState={setEditState}
            canRemove={canRemove}
            canEdit={canEdit}
            canLock={canLock}
          />
        </>
      );
      break;
    case 'LOCKED':
      buttons = (
        <>
          <LockedPaymentPlanHeaderButtons
            paymentPlan={paymentPlan}
            canDuplicate={canDuplicate}
            canLock={canLock}
            canSendForApproval={canSendForApproval}
          />
        </>
      );
      break;
    case 'IN_APPROVAL':
      buttons = (
        <>
          <InApprovalPaymentPlanHeaderButtons
            paymentPlan={paymentPlan}
            canReject={canReject}
            canApprove={canApprove}
          />
        </>
      );
      break;
    case 'IN_AUTHORIZATION':
      buttons = (
        <>
          <InAuthorizationPaymentPlanHeaderButtons
            paymentPlan={paymentPlan}
            canReject={canReject}
            canAuthorize={canAuthorize}
          />
        </>
      );
      break;
    case 'IN_REVIEW':
      buttons = (
        <>
          <InReviewPaymentPlanHeaderButtons
            paymentPlan={paymentPlan}
            canReject={canReject}
            canMarkAsReviewed={canMarkAsReviewed}
          />
        </>
      );
      break;
    case 'ACCEPTED':
      buttons = (
        <>
          <AcceptedPaymentPlanHeaderButtons
            paymentPlan={paymentPlan}
            canDownloadXlsx={canDownloadXlsx}
            canSendToFsp={canSendToFsp}
          />
        </>
      );
      break;
    default:
      buttons = <></>;
      break;
  }

  return (
    <PageHeader
      title={
        <Box display='flex' alignItems='center'>
          {t('Payment Plan')} ID ${paymentPlan.unicefId}
          <StatusWrapper>
            <StatusBox
              status={paymentPlan.status}
              statusToColor={paymentPlanStatusToColor}
              statusNameMapping={paymentPlanStatusMapping}
            />
          </StatusWrapper>
        </Box>
      }
      breadCrumbs={
        hasPermissions(PERMISSIONS.PAYMENT_MODULE_VIEW_DETAILS, permissions)
          ? breadCrumbsItems
          : null
      }
    >
      {buttons}
    </PageHeader>
  );
};
