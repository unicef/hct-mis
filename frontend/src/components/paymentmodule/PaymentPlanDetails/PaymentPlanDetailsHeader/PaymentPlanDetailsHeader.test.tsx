import React from 'react';
import { PERMISSIONS } from '../../../../config/permissions';
import { render } from '../../../../testUtils/testUtils';
import { CreatePaymentPlanHeader } from './PaymentPlanDetailsHeader';

describe('components/paymentmodule/CreatePaymentPlanHeader', () => {
  it('should render', () => {
    const { container } = render(
      <CreatePaymentPlanHeader
        handleSubmit={jest.fn()}
        businessArea='afghanistan'
        permissions={[PERMISSIONS.PAYMENT_MODULE_VIEW_LIST]}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
