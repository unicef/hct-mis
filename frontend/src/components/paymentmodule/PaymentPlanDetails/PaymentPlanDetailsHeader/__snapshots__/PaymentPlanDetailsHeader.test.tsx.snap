// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsHeader/MarkAsReleasedPaymentPlan should render 1`] = `
<div>
  <div
    class="MuiBox-root css-hpgf8j"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-dl1jos-MuiButtonBase-root-MuiButton-root"
      data-cy="button-mark-as-released"
      disabled=""
      tabindex="-1"
      type="button"
    >
      Mark as released
    </button>
  </div>
</div>
`;
